script.js

```diff
@@ -1,4 +1,408 @@
+    },
+    {
+      name: "6",
+      suit: "spades",
+      rank: 6,
+    },
+    {
+      name: "7",
+      suit: "spades",
+      rank: 7,
+    },
+    {
+      name: "8",
+      suit: "spades",
+      rank: 8,
+    },
+    {
+      name: "9",
+      suit: "spades",
+      rank: 9,
+    },
+    {
+      name: "10",
+      suit: "spades",
+      rank: 10,
+    },
+    {
+      name: "jack",
+      suit: "spades",
+      rank: 10,
+    },
+    {
+      name: "queen",
+      suit: "spades",
+      rank: 10,
+    },
+    {
+      name: "king",
+      suit: "spades",
+      rank: 10,
+    },
+  ];
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+var pickCard = function () {
+  card = cardDeck.pop();
+  return card;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //shuffle the deck with the first submit click
+  if (gameMode == "shuffle") {
+    shuffledDeck = shuffleCards(cardDeck);
+    console.log(shuffledDeck);
+    gameMode = "player pick";
+    return "The cards have been shuffled and the game is ready to begin!";
+  }
+  //initial player pick gets displayed and the count is recorded
+  if (gameMode == "player pick") {
+    gameMode = "player further picks or computer";
+    playerCard1 = pickCard(shuffledDeck);
+    playerCard2 = pickCard(shuffledDeck);
+    //case when player gets an ace
+    if (playerCard1.rank == 1 && playerCard2.rank < 11) {
+      playerCard1.rank = 11;
+    }
+    if (playerCard2.rank == 1 && playerCard1.rank < 11) {
+      playerCard2.rank = 11;
+    }
+    playerCount = playerCard1.rank + playerCard2.rank;
+    console.log(playerCount);
+    return `Your first two cards are ${playerCard1.name} of ${playerCard1.suit} and ${playerCard2.name} of ${playerCard2.suit}. Your total count sits at ${playerCount}<br><br>If you want to pick another card please click 'submit', else please input 'computer' into the input box to let the computer play!`;
+  }
+  //player picks another card
+  if (gameMode == "player further picks or computer" && input == "") {
+    playerCard1 = pickCard(shuffledDeck);
+    //case when player gets an ace
+    if (playerCard1.rank == 1 && playerCount < 11) {
+      playerCard1.rank = 11;
```

Comment:
maybe we could make ace condition into a function and reuse it instead of typing the same logic